go.mod

```diff
@@ -1,7 +1,9 @@
 module go.pinniped.dev
 
 // k8s.io/api@v0.29.0 requires go@1.21
-go 1.21
+go 1.21.3
+
+toolchain go1.22.0
```

Comment:
Hmm, I don't think we want this change.

Reply:
One of our direct deps now says `go 1.21.3` in its go.mod file. https://github.com/kubernetes/kube-openapi/blob/f107216b40e23161bb0fcfe35456c10cbd63b9c6/go.mod#L3

This was a very recent change in kube-openapi, and the diff can be seen here: https://github.com/kubernetes/kube-openapi/commit/b9a9bc3efda215e80efe28641d3cd07c097722f6

According to https://go.dev/doc/toolchain:

> "The go get and go mod tidy commands maintain the go line to be greater than or equal to the go line of any required dependency module."

So it seems that if we want to update to the latest version of our kube-openapi dep, then we have no choice but to inherit its `go` declaration.

Reply:
According to https://go.dev/ref/mod#go-mod-file-toolchain:

> "For reproducibility, the go command writes its own toolchain name in a toolchain line any time it is updating the go version in the go.mod file (usually during go get)."

So we apparently have no choice but to start putting the `toolchain` directive into our go.mod as well, unless we want to figure out how to explicitly override it.

It picked `1.22.0` because that is the version of Go that we are currently telling CI to use when running our dependency bumping automation tool.

Using 1.22.0 seems like a reasonable choice, because that is what is declared in our Dockerfile so that is what CI will use when building the server container image. It is also the version that CI is currently using to build the CLI binaries, build the test binaries, run the tests, etc.

Although the project currently should work fine with the older 1.21.x also (I think, we don't fully test this in CI), it seems like it would not hurt to add this `toolchain` directive to effectively let the rest of the world know the version of Go that we actually use for our development work. The `toolchain` directive "declares a suggested Go toolchain to use with a module", so it apparently does not prevent someone from using Go 1.21 if they really want to try that for some reason. They could change their `GOTOOLCHAIN` environment variable as described in https://go.dev/doc/toolchain.

Reply:
For some reason, this breaks the "CodeQL / Analyze (go)" check on the PR. Its output includes this error message: `The go.mod file requires version v1.21.3 of Go, but version v1.20.14 is installed. Consider adding an actions/setup-go step to your workflow.`.

Reply:
I added a new commit to this branch to attempt to resolve the problem with CodeQL by changing its workflow yaml file. I upgraded all our steps to use the latest versions and added a new step for `actions/setup-go`. I also changed this repo's GitHub settings to allow actions `actions/setup-go@*`.